zerver/lib/create_user.py

```diff
@@ -115,9 +116,13 @@ def create_user_profile(
     )
     if bot_type or not active:
         password = None
-    if user_profile.email_address_is_realm_public():
-        # If emails are visible to everyone, we can set this here and save a DB query
-        user_profile.email = get_display_email_address(user_profile)
+    if (
+        bot_type
+        or realm_user_default.email_address_visibility
+        == RealmUserDefault.EMAIL_ADDRESS_VISIBILITY_EVERYONE
+    ):
```

Comment:
Did we need to replace `user_profile.email_address_is_realm_public` with this more complex conditional?  We still use it in this file, and I think the value is unchanged from before.

Oh, I guess the issue is that we haven't set `user_profile.email_address_visibility yet`?  

Maybe it'd be a cleaner interface to just pass `email_address_visibility` into this function from the caller, rather than the whole `RealmUserDefault` object, set it in the `UserProfile` invocation (with a comment that we need to do this early for the stated reason), and then have this call `user_profile.email_address_is_realm_public` as before.